manual_flagging/flagging.py

```diff
@@ -0,0 +1,517 @@
+                    labels=char_bldg_sf_labels,
+                )
+                # Bin age
+                bldg_age_bins, bldg_age_labels = create_bins_and_labels(
+                    inputs["stat_groups"]["current"][f"tri{tri}"]["res_single_family"][
+                        "age_bin_specification"
+                    ]
+                )
+                df_copy["bldg_age_bin"] = pd.cut(
+                    df_copy["bldg_age"],
+                    bins=bldg_age_bins,
+                    labels=bldg_age_labels,
+                )
+                dfs_to_rolling_window[key]["df"] = df_copy
+
+            # Binning feature creation
+            if "res_multi_family" in key:
+                df_copy = dfs_to_rolling_window[key]["df"].copy()
+
+                # Define bins for building age
+                bldg_age_bins, bldg_age_labels = create_bins_and_labels(
+                    inputs["stat_groups"]["current"][f"tri{tri}"]["res_multi_family"][
+                        "age_bin_specification"
+                    ]
+                )
+                df_copy["bldg_age_bin"] = pd.cut(
+                    df_copy["bldg_age"],
+                    bins=bldg_age_bins,
+                    labels=bldg_age_labels,
+                )
+                dfs_to_rolling_window[key]["df"] = df_copy
+
+        if tri == 2:
+            # Currently have not developed a new method for tri 2
+            pass
+        if tri == 3:
+            # Currently have not developed a new method for tri 3
+            pass
+
+# Collect tris that will be flagged for OG method
+og_mansueto_tris = [
+    tri for tri, method in tri_stat_groups.items() if method == "og_mansueto"
+]
+
+if len(og_mansueto_tris) != 0:
+    # Filter by triad code and market type
+    og_mansueto_filter = ~df["triad_code"].isin(og_mansueto_tris)
+
+    df_res_og_mansueto = df[(df["indicator"] == "res") & triad_code_filter]
+    df_condo_og_mansueto = df[(df["indicator"] == "condo") & triad_code_filter]
+
+    # Append these DataFrames to the dictionary
+    key_res = (
+        f"df_tri{'&'.join(str(number) for number in og_mansueto_tris)}_res_og_mansueto"
+    )
+    key_condo = f"df_tri{'&'.join(str(number) for number in og_mansueto_tris)}_condos_og_mansueto"
+
+    dfs_to_rolling_window[key_res] = {
+        "df": df_res_og_mansueto,
+        "columns": inputs["stat_groups"]["og_mansueto"]["res_single_family"]["columns"],
+        "iso_forest_cols": inputs["iso_forest"]["res"],
+        "condos_boolean": False,
+        "market": "res_og_mansueto",
+    }
+    dfs_to_rolling_window[key_condo] = {
+        "df": df_condo_og_mansueto,
+        "columns": inputs["stat_groups"]["og_mansueto"]["condos"]["columns"],
+        "iso_forest_cols": inputs["iso_forest"]["condos"],
+        "condos_boolean": True,
+        "market": "condos_og_mansueto",
+    }
+
+# - - - - - -
+# Make rolling window
+# - - - - - -
+
+dfs_to_flag = copy.deepcopy(dfs_to_rolling_window)
```

Comment:
Going to take a stab at replying here before we hop on a call. Let me know if I misunderstood your question. I see two drawbacks with instantiating an empty dictionary instead of using `deepcopy` here. 

The first is that copying the dict gives us all of the auxiliary data for each dataframe.
```python
# - - - - -
# Flag Sales
# - - - - -

dfs_flagged = copy.deepcopy(dfs_to_flag)

for df_name, df in dfs_to_flag.items():
    print(f"\nFlagging sales for {df_name}")
    df_copy = df["df"].copy()
    df_copy = flg_model.go(
        df=df_copy,
        groups=tuple(df["columns"]),
        iso_forest_cols=df["iso_forest_cols"],
        dev_bounds=tuple(inputs["dev_bounds"]),
        condos=df["condos_boolean"],
    )

    # Add the edited or unedited dataframe to the new dictionary
    dfs_flagged[df_name]["df"] = df_copy
```